src/api/v4/dataview/time-series/index.js

```diff
@@ -0,0 +1,177 @@
+ * 
+ * @return {carto.dataview.timeAggregation} Current time aggregation
+ * @api 
+ */
+TimeSeries.prototype.getAggregation = function () {
+  return this._aggregation;
+};
+
+/**
+ * Set time offset
+ * 
+ * @param {number} offset
+ * @return {carto.dataview.TimeSeries} this
+ * @api
+ */
+TimeSeries.prototype.setOffset = function (offset) {
+  this._validateOffset(offset);
+  var prevOffset = this._offset;
+  this._offset = offset;
+  if (this._internalModel) {
+    this._internalModel.set('offset', hoursToSeconds(offset));
+  } else if (prevOffset !== offset) {
+    this._triggerChange('offset', offset);
+  }
+  return this;
+};
+
+/**
+ * Return the current time offset
+ * 
+ * @return {number} Current time offset
+ * @api 
+ */
+TimeSeries.prototype.getOffset = function () {
+  return this._offset;
+};
+
+/**
+ * Set the local timezone flag. If enabled, the time offset is overriden by the use local timezone
+ * 
+ * @param {boolean} localTimezone
+ * @return {carto.dataview.TimeSeries} this
+ * @api
+ */
+TimeSeries.prototype.setLocalTimezone = function (localTimezone) {
+  this._validateLocalTimezone(localTimezone);
+  this._changeProperty('localTimezone', localTimezone);
+  return this;
+};
+
+/**
+ * Return the current local timezone flag
+ * 
+ * @return {boolean} Current local timezone flag
+ * @api 
+ */
+TimeSeries.prototype.getLocalTimezone = function () {
+  return this._localTimezone;
+};
+
+TimeSeries.prototype._checkOptions = function (options) {
+  if (_.isUndefined(options)) {
+    throw new TypeError('Options object to create a histogram dataview is required.');
+  }
+  this._validateAggregation(options.aggregation);
+  this._validateOffset(options.offset);
+  this._validateLocalTimezone(options.localTimezone);
+};
+
+TimeSeries.prototype._validateAggregation = function (aggregation) {
+  if (!isValidTimeAggregation(aggregation)) {
+    throw new TypeError('Time aggregation must be a valid value. Use carto.dataview.timeAggregation.');
+  }
+};
+
+TimeSeries.prototype._validateOffset = function (offset) {
+  if (!_.isFinite(offset) || Math.floor(offset) !== offset || offset < -12 || offset > 14) {
+    throw new TypeError('Offset must an integer value between -12 and 14.');
+  }
+};
+
+TimeSeries.prototype._validateLocalTimezone = function (localTimezone) {
+  if (!_.isBoolean(localTimezone)) {
+    throw new TypeError('LocalTimezone must be a boolean value.');
+  }
+};
+
+TimeSeries.prototype._listenToInternalModelSpecificEvents = function () {
+  // Empty function
+};
+
+TimeSeries.prototype._createInternalModel = function (engine) {
+  this._internalModel = new HistogramDataviewModel({
+    source: this._source.$getInternalModel(),
+    column: this._column,
+    aggregation: this._aggregation,
+    offset: hoursToSeconds(this._offset),
+    localTimezone: this._localTimezone,
+    sync_on_data_change: true,
+    sync_on_bbox_change: !!this._boundingBoxFilter,
+    enabled: this._enabled,
+    column_type: 'date'
```

Comment:
Would be nice to have a constant for Dataview column types instead of using 'date'...

Reply:
Well, this is internal to be able to use `histogram` dataview as if it was two separate entities. Not sure if it's worthy in this case.